src/main/java/com/google/codeu/data/Message.java

```diff
@@ -26,25 +26,31 @@
   private String text;
   private long timestamp;
   private String imageUrl;
+  private String conversationTopicId;
 
   /**
    * Constructs a new {@link Message} posted by {@code user} with {@code text} content. Generates a
    * random ID and uses the current system time for the creation time.
    */
   public Message(String user, String text) {
-    this(UUID.randomUUID(), user, text, System.currentTimeMillis(), null); // imageUrl is null if user didn't upload an image
+    this(UUID.randomUUID(), user, text, System.currentTimeMillis(), null, null); // imageUrl is null if user didn't upload an image
   }
   
   public Message(String user, String text, String imageUrl) {
-	  this(UUID.randomUUID(), user, text, System.currentTimeMillis(), imageUrl);
+	  this(UUID.randomUUID(), user, text, System.currentTimeMillis(), imageUrl, null);
   }
 
-  public Message(UUID id, String user, String text, long timestamp, String imageUrl) {
+  public Message(String user, String text, String imageUrl, String conversationTopicId) {
+    this(UUID.randomUUID(), user, text, System.currentTimeMillis(), imageUrl, conversationTopicId);
+  }
+
+  public Message(UUID id, String user, String text, long timestamp, String imageUrl, String conversationTopicId) {
     this.id = id;
     this.user = user;
     this.text = text;
     this.timestamp = timestamp;
     this.imageUrl = imageUrl;
+    this.conversationTopicId = conversationTopicId;
```

Comment:
Not all messages will have conversationTopicId, right?
You may want to create an overloaded constructor.
Ideally, you could use the builder pattern: good tutorial here https://dzone.com/articles/design-patterns-the-builder-pattern